sycl/source/detail/scheduler/commands.cpp

```diff
@@ -3511,6 +3511,17 @@ ur_result_t ExecCGCommand::enqueueImpQueue() {
     const AdapterPtr &Adapter = MQueue->getAdapter();
     if (MEvent != nullptr)
       MEvent->setHostEnqueueTime();
+    // Honor explicit dependencies set with depends_on.
+    // Important for cross queue dependencies.
```

Comment:
Would the following be accurate:

"The nature of a barrier makes it kind of work if we don't handle these dependencies if they belong to the same queue, but the issue (re-)appears if they're coming from another queue"?

Reply:
https://github.com/intel/llvm/pull/18139/commits/c64a71a6df9418858f95d604982728ed6e301136